alkira/resource_alkira_service_pan.go

```diff
@@ -333,22 +333,29 @@ func resourceServicePanDelete(d *schema.ResourceData, m interface{}) error {
 	return client.DeleteServicePan(d.Id())
 }
 
-func generateServicePanRequest(d *schema.ResourceData, m interface{}) (*alkira.ServicePan, error) {
+func generateServicePanRequest(d *schema.ResourceData, gs getSegmentById) (*alkira.ServicePan, error) {
 
-	billingTagIds := convertTypeListToIntList(d.Get("billing_tag_ids").([]interface{}))
-	instances := expandPanInstances(d.Get("instance").(*schema.Set))
 	panoramaDeviceGroup := d.Get("panorama_device_group").(string)
 	panoramaIpAddresses := convertTypeListToStringList(d.Get("panorama_ip_addresses").([]interface{}))
 	panoramaTemplate := d.Get("panorama_template").(string)
-	segmentIds := convertTypeListToIntList(d.Get("segment_ids").([]interface{}))
-	segmentOptions := expandPanSegmentOptions(d.Get("zones_to_groups").(*schema.Set))
 
-	globalProtectSegmentOptions := expandGlobalProtectSegmentOptions(
-		d.Get("global_protect_segment_options").(*schema.Set),
-	)
+	instances, err := expandPanInstances(d.Get("instance").(*schema.Set), gs)
+	if err != nil {
+		return nil, err
+	}
+
+	segmentOptions, err := expandPanSegmentOptions(d.Get("zones_to_groups").(*schema.Set), gs)
+	if err != nil {
+		return nil, err
+	}
+
+	globalProtectSegmentOptions, err := expandGlobalProtectSegmentOptions(d.Get("global_protect_segment_options").(*schema.Set), gs)
+	if err != nil {
+		return nil, err
+	}
 
 	service := &alkira.ServicePan{
-		BillingTagIds:               billingTagIds,
+		BillingTagIds:               convertTypeListToIntList(d.Get("billing_tag_ids").([]interface{})),
```

Comment:
why this is related to segment change?

Reply:
It's just a simple cleanup. This was defined as a variable above. Since no error can be returned I assigned it in the struct directly.